src/plugins/ui_feature/renderer.ts

```diff
@@ -144,24 +145,72 @@ export class RendererPlugin extends UIPlugin {
     const areGridLinesVisible =
       !this.getters.isDashboard() &&
       this.getters.getGridLinesVisibility(this.getters.getActiveSheetId());
-    const inset = areGridLinesVisible ? 0.1 * thinLineWidth : 0;
 
     if (areGridLinesVisible) {
       for (const box of this.boxes) {
         ctx.strokeStyle = CELL_BORDER_COLOR;
         ctx.lineWidth = thinLineWidth;
-        ctx.strokeRect(box.x + inset, box.y + inset, box.width - 2 * inset, box.height - 2 * inset);
+        ctx.strokeRect(box.x, box.y, box.width, box.height);
       }
     }
   }
 
   private drawCellBackground(renderingContext: GridRenderingContext) {
-    const { ctx } = renderingContext;
+    const { ctx, thinLineWidth } = renderingContext;
+    const areGridLinesVisible =
+      !this.getters.isDashboard() &&
+      this.getters.getGridLinesVisibility(this.getters.getActiveSheetId());
     for (const box of this.boxes) {
       let style = box.style;
       if (style.fillColor && style.fillColor !== "#ffffff") {
         ctx.fillStyle = style.fillColor || "#ffffff";
-        ctx.fillRect(box.x, box.y, box.width, box.height);
+        ctx.fillRect(
+          box.x - thinLineWidth / 2,
+          box.y - thinLineWidth / 2,
+          box.width + thinLineWidth,
+          box.height + thinLineWidth
+        );
+        if (areGridLinesVisible) {
```

Comment:
drawing grid lines seems scattered in several places. There's one part in `drawBackground` and there some here now.
Could it be moved all together somewhere ?
Maybe a new method `drawGridLines` ?

Reply:
> drawing grid lines seems scattered in several places. There's one part in `drawBackground` and there some here now. Could it be moved all together somewhere ? Maybe a new method `drawGridLines` ?

@LucasLefevre I am not sure what you mean "moved all together somewhere". Do you mean:
1. extract these three lines of code into a new function and call this new function in each place 
OR 
2. draw these grid lines all in once together with the drawing in `drawBackground`? 

If it is 1: sure!
If it is 2: I don't think the "gridlines" in `drawCellBackground` (gridlines 1) can be drawn together with the gridlines in `drawBackground` (gridlines 2). Gridlines1 is used to differentiate the borders between the user-defined background color and the gridlines 2 (the target of this task), so it needs to be drawn above the user-defined background color and the overflowing background (which is used to eliminate the overflowing cell separators). Gridlines2 need to be drawn under the overflowing background, otherwise the overflowing background doesn't work.